Practical 4 Task- About Looping, Switch and static concepts.

Jay-Prajapat opened the pull request:
In this task : Implemented CalculateAverageMarks method, CalculateGrade method, FindMaximumMarks and FindMinimumMarks method.

Jay-Prajapat committed:
Practical 4 Task- About Looping, Switch and static concepts.

In this task : Implemented CalculateAverageMarks method, CalculateGrade method, FindMaximumMarks and FindMinimumMarks method.

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
```
Install linting and formatting tools in .net so that it is properly formatted code

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+    {
+        static void Main(string[] args)
+        {
+            Student student1 = new Student();
```
Student1 is not accepted

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+            {
+                sum += marks[i];
+            }
+            AverageMarks = sum / marks.Length;
```
always null check the variable before accessing .length

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+            student1.Name = Console.ReadLine();
+            Console.WriteLine("Enter marks of student.");
+            for (int i = 0; i < 5; i++)
+            {
```
There should be some validation between which range user can add the marks

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+        {
+            string grade = "";
+            switch (mark)
+            {
```
switch case will not handle the scenario when it is exact 70,80,90 etc

dilaara-simform commented on Practical4/Practical4/Program.cs:
```diff
@@ -0,0 +1,130 @@
+                        break;
+                    case (int)Student.Options.Grade:
+                        decimal averageMarks = student1.CalculateAverageMarks(student1.Marks);
+                        Console.WriteLine($"The Grade is {student1.CalculateGrade(averageMarks)}");
```
you need to check whether you are getting response from calcualteGrade function then only need to print the console.writeline.

Jay-Prajapat committed:
Made changes in program.cs file.

-Given proper name to instance variable.
-Checked the null value of the variable before accessing length.
-Added the validation for accessing marks.
-Checked the value of calculateGrade method before printing.
